test_api_interfaces.TestApiVxlanInterface fails on DUTs running 4.15 and higher

If you run the suite of system tests on a DUT running EOS version 4.15.0 or higher the test_get in the TestApiVxlanInterface class fails:

```
FAIL: test_get (test_api_interfaces.TestApiVxlanInterface)
Traceback (most recent call last):
File "/Users/dt/eosplus/pyeapi/test/system/test_api_interfaces.py", line 343, in test_get
self.assertEqual(result['multicast_group'], '')
AssertionError: 'decap' != ''
```

This is due to a new EOS command with the following syntax:
`[no] vxlan multicast-group decap`

Therefore the current regex in `_parse_multicast_group` will match on decap when it tries to parse through show running-config all interface vxlan.  The resolution to this issue will be to update the regex in `_parse_multicast_group` to match on a correctly formatted IPv4 multicast group.

--- a/pyeapi/api/interfaces.py
+++ b/pyeapi/api/interfaces.py
@@ -756,7 +756,7 @@ class VxlanInterface(BaseInterface):
         return dict(source_interface=value)
 
     def _parse_multicast_group(self, config):
-        match = re.search(r'vxlan multicast-group ([\d]{3}\.[\d]+.[\d]+.[\d]+)', 
+        match = re.search(r'vxlan multicast-group ([\d]{3}\.[\d]+\.[\d]+\.[\d]+)', 
                           config)
         value = match.group(1) if match else self.DEFAULT_MCAST_GRP
         return dict(multicast_group=value)

--- a/pyeapi/api/interfaces.py
+++ b/pyeapi/api/interfaces.py
@@ -710,6 +710,8 @@ class VxlanInterface(BaseInterface):
             * udp_port (int): The vxlan udp-port value
             * vlans (dict): The vlan to vni mappings
             * flood_list (list): The list of global VTEP flood list
+            * multicast_decap (bool): If the mutlicast decap 
+                                      feature is configured
 
         Args:
             name (str): The interface identifier to retrieve from the
@@ -732,6 +734,7 @@ class VxlanInterface(BaseInterface):
         response.update(self._parse_udp_port(config))
         response.update(self._parse_vlans(config))
         response.update(self._parse_flood_list(config))
+        response.update(self._parse_multicast_decap(config))
 
         return response
 
@@ -753,9 +756,14 @@ class VxlanInterface(BaseInterface):
         return dict(source_interface=value)
 
     def _parse_multicast_group(self, config):
-        match = re.search(r'vxlan multicast-group ([^\s]+)', config)
+        match = re.search(r'vxlan multicast-group ([\d]{3}\.[\d]+.[\d]+.[\d]+)', 
+                          config)
         value = match.group(1) if match else self.DEFAULT_MCAST_GRP
         return dict(multicast_group=value)
+   
+    def _parse_multicast_decap(self, config):
+        value = 'vxlan mutlicast-group decap' in config
+        return dict(multicast_decap=bool(value))
 
     def _parse_udp_port(self, config):
         match = re.search(r'vxlan udp-port (\d+)', config)
@@ -830,6 +838,31 @@ class VxlanInterface(BaseInterface):
         cmds = self.command_builder(string, value=value, default=default,
                                     disable=disable)
         return self.configure_interface(name, cmds)
+
+    def set_multicast_decap(self, name, default=False,
+                           disable=False):
+        """Configures the Vxlan multicast-group decap feature 
+
+        EosVersion:
+            4.15.0M
+
+        Args:
+            name(str): The interface identifier to configure, defaults to
+                Vxlan1
+           default(bool): Configures the mulitcast-group decap value to default
+           disable(bool): Negates the multicast-group decap value
+
+        Returns:
+            True if the operation succeeds otherwise False
+        """
+        string = 'vxlan multicast-group decap'
+        if(default or disable):
+            cmds = self.command_builder(string, value=None, default=default,
+                                       disable=disable)
+        else:
+            cmds = [string] 
+        return self.configure_interface(name, cmds)
+
 
     def set_udp_port(self, name, value=None, default=False, disable=False):
         """Configures vxlan udp-port value

--- a/test/fixtures/running_config.vxlan
+++ b/test/fixtures/running_config.vxlan
@@ -46,4 +46,5 @@ interface Vxlan1
    no vxlan vlan flood vtep
    no vxlan learn-restrict vtep
    no vxlan vlan learn-restrict vtep
+   no vxlan multicast-group decap
 !

--- a/test/fixtures/vxlan.json
+++ b/test/fixtures/vxlan.json
@@ -9,6 +9,7 @@
             "vlanToVtepList": {},
             "mtu": 0,
             "hardware": "vxlan",
+            "mcastGrpDecap": "",
             "replicationMode": "multicast",
             "bandwidth": 0,
             "floodMcastGrp": "239.10.10.10",

--- a/test/system/test_api_interfaces.py
+++ b/test/system/test_api_interfaces.py
@@ -341,6 +341,8 @@ class TestApiVxlanInterface(DutSystemTest):
             self.assertEqual(result['description'], None)
             self.assertEqual(result['source_interface'], '')
             self.assertEqual(result['multicast_group'], '')
+            self.assertEqual(result['multicast_decap'], False)
+
 
     def get_config(self, dut):
         cmd = 'show running-config all interfaces Vxlan1'
@@ -404,6 +406,16 @@ class TestApiVxlanInterface(DutSystemTest):
             instance = api.set_multicast_group('Vxlan1', disable=True)
             self.assertTrue(instance)
             self.contains('no vxlan multicast-group', dut)
+
+    '''commenting this one out as it will only parse on a  trident based DUT
+    def test_set_multicast_decap(self):
+        for dut in self.duts:
+            dut.config(['no interface Vxlan1', 'interface Vxlan1'])
+            api = dut.api('interfaces')
+            instance = api.set_multicast_decap('Vxlan1')
+            self.assertTrue(instance)
+            self.contains('vxlan multicast-group decap', dut)
+    '''
 
     def test_set_udp_port(self):
         for dut in self.duts:

--- a/test/system/test_client.py
+++ b/test/system/test_client.py
@@ -71,9 +71,8 @@ class TestClient(unittest.TestCase):
 
     def test_no_enable_single_command_no_auth(self):
         for dut in self.duts:
-            dut.run_commands('disable')
             with self.assertRaises(pyeapi.eapilib.CommandError):
-                dut.run_commands('show running-config', 'json', send_enable=False)
+                dut.run_commands(['disable', 'show running-config'], 'json', send_enable=False)
 
     def test_enable_multiple_commands(self):
         for dut in self.duts:
@@ -187,7 +186,7 @@ class TestNode(unittest.TestCase):
         # Send a continuous command that requires a break
         cases.append(('watch 10 show int e1 count rates', rfmt
                       % (1000, 'could not run command',
-                         'init error \(cbreak\(\) returned ERR\)')))
+                         'init error.*')))
         # Send a command that has insufficient priv
         cases.append(('show running-config', rfmt
                       % (1002, 'invalid command',

--- a/test/unit/test_api_interfaces.py
+++ b/test/unit/test_api_interfaces.py
@@ -407,8 +407,6 @@ class TestApiVxlanInterface(EapiConfigUnitTest):
         func = function('set_multicast_decap', 'Vxlan1', default=True)
         self.eapi_positive_config_test(func, cmds)
 
-
-
     def test_set_udp_port_with_value(self):
         cmds = ['interface Vxlan1', 'vxlan udp-port 1024']
         func = function('set_udp_port', 'Vxlan1', '1024')

--- a/test/unit/test_api_interfaces.py
+++ b/test/unit/test_api_interfaces.py
@@ -357,7 +357,8 @@ class TestApiVxlanInterface(EapiConfigUnitTest):
 
     def test_get(self):
         keys = ['name', 'type', 'description', 'shutdown', 'source_interface',
-                'multicast_group', 'udp_port', 'vlans', 'flood_list']
+                'multicast_group', 'udp_port', 'vlans', 'flood_list',
+                'multicast_decap']
         result = self.instance.get('Vxlan1')
         self.assertEqual(sorted(keys), sorted(result.keys()))
 
@@ -390,6 +391,23 @@ class TestApiVxlanInterface(EapiConfigUnitTest):
         cmds = ['interface Vxlan1', 'default vxlan multicast-group']
         func = function('set_multicast_group', 'Vxlan1', default=True)
         self.eapi_positive_config_test(func, cmds)
+
+    def test_set_multicast_decap(self):
+        cmds = ['interface Vxlan1', 'vxlan multicast-group decap']
+        func = function('set_multicast_decap', 'Vxlan1')
+        self.eapi_positive_config_test(func, cmds)
+
+    def test_set_multicast_decap_with_no_value(self):
+        cmds = ['interface Vxlan1', 'no vxlan multicast-group decap']
+        func = function('set_multicast_decap', 'Vxlan1', disable=True)
+        self.eapi_positive_config_test(func, cmds)
+
+    def test_set_multicast_decap_with_default(self):
+        cmds = ['interface Vxlan1', 'default vxlan multicast-group decap']
+        func = function('set_multicast_decap', 'Vxlan1', default=True)
+        self.eapi_positive_config_test(func, cmds)
+
+
 
     def test_set_udp_port_with_value(self):
         cmds = ['interface Vxlan1', 'vxlan udp-port 1024']

